### PR TITLE
Doc: 13.1 changelog

### PR DESCRIPTION
**Pull request by 2TallTyler**
## Motivation / Problem

We should release 13.1.

## Description

Update changelog with all backported PRs since 13.0.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,25 @@
+13.1 (2023-04-10)
+------------------------------------------------------------------------
+Feature: [NewGRF] Engine name callback. (#10399)
+Change: Avoid crashing to the side of a train (#10496)
+Fix 8361cf5a73: Missing bounds check for house specs. (#10625)
```
Comment (LordAro) [inline changelog.txt]: hash

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,25 @@
...
+Fix: Clicking on editbox clear button didn't take account of padding. (#10583)
+Fix: [Script] Access to enum/consts defined outside of main.nut (#10573)
+Fix #10568: "can savegame be loaded check" failed in dedicated server (#10572)
+Fix #10554: Let Scrollbar::SetPosition clamp instead of assert. (#10555)
+Fix: Network server highlight invisible with RTL layout. (#10551)
+Fix: Restore using founder client name as company manager name (#10535)
+Fix: Don't send unused tile field over the network (#10507)
+Fix #10477: Not enough space for text due to rounding down (OSX) (#10489)
+Fix #10486: [Script] Debug window requires AIs to be started before GS (#10487)
+Fix #10469, 5e14a20: [Script] League Table rating element is a int64 everywhere else (#10471)
```
Comment (LordAro) [inline changelog.txt]: hash

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,25 @@
+13.1 (2023-04-10)
+------------------------------------------------------------------------
+Feature: [NewGRF] Engine name callback. (#10399)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Feature: [NewGRF] Engine name callback for variant purposes (#10399)
```

Not ideal, but feels like there should be some more context

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,25 @@
...
+Change: Avoid crashing to the side of a train (#10496)
```
Comment (LordAro) [inline changelog.txt]: Probably say this was a fix

```suggestion
Fix: Prevent road vehicles on crossing from crashing into the side of a train (#10496)
```

File: changelog.txt (hunk 1 above)
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix [NewGRF]: Missing bounds check for House GRFs (#10625)
```

possibly minor enough to exclude, or lump in with some of the others

File: changelog.txt [hunk 5]
```diff
@@ -1,3 +1,25 @@
...
+Fix 8361cf5a73: Missing bounds check for house specs. (#10625)
+Fix: Update top toolbar tooltips for added features. (#10616)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix: Improve main toolbar tooltips (#10616)
```

File: changelog.txt [hunk 6]
```diff
@@ -1,3 +1,25 @@
...
+Fix: NewGRF Action3 (+others) validation (#10601)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix [NewGRF]: Additional validation for Action3 (+others) (#10601)
```

File: changelog.txt [hunk 7]
```diff
@@ -1,3 +1,25 @@
...
+Fix: Clicking on editbox clear button didn't take account of padding. (#10583)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix: Clear button for editbox didn't take account of padding (#10583)
```

File: changelog.txt [hunk 8]
```diff
@@ -1,3 +1,25 @@
...
+Fix: [Script] Access to enum/consts defined outside of main.nut (#10573)
+Fix #10568: "can savegame be loaded check" failed in dedicated server (#10572)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10568: Bogus warning when loading a save with a NewGRFs on dedicated servers (#10572)
```

File: changelog.txt [hunk 9]
```diff
@@ -1,3 +1,25 @@
...
+Fix #10554: Let Scrollbar::SetPosition clamp instead of assert. (#10555)
```
Comment (LordAro) [inline changelog.txt]: Fix #10554: Crash when scrolling in the autoreplace window with collapsed variants (#10555)

File: changelog.txt [hunk 10]
```diff
@@ -1,3 +1,25 @@
...
+Fix: Network server highlight invisible with RTL layout. (#10551)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix: Network server highlight invisible with RTL languages (#10551)
```

File: changelog.txt [hunk 11]
```diff
@@ -1,3 +1,25 @@
...
+Fix: Restore using founder client name as company manager name (#10535)
+Fix: Don't send unused tile field over the network (#10507)
```
Comment (LordAro) [inline changelog.txt]: Remove, not user visible(?)

File: changelog.txt [hunk 12]
```diff
@@ -1,3 +1,25 @@
...
+Change: Avoid crashing to the side of a train (#10496)
+Fix 8361cf5a73: Missing bounds check for house specs. (#10625)
+Fix: Update top toolbar tooltips for added features. (#10616)
+Fix: NewGRF Action3 (+others) validation (#10601)
+Fix: Clicking on editbox clear button didn't take account of padding. (#10583)
+Fix: [Script] Access to enum/consts defined outside of main.nut (#10573)
+Fix #10568: "can savegame be loaded check" failed in dedicated server (#10572)
+Fix #10554: Let Scrollbar::SetPosition clamp instead of assert. (#10555)
+Fix: Network server highlight invisible with RTL layout. (#10551)
+Fix: Restore using founder client name as company manager name (#10535)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix: Client name was not being used as company manager name (#10535)
```

File: changelog.txt [hunk 13]
```diff
@@ -1,3 +1,25 @@
...
+Fix: Don't send unused tile field over the network (#10507)
+Fix #10477: Not enough space for text due to rounding down (OSX) (#10489)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10477: [macOS] Calculation for window sizes when using custom fonts was being rounded incorrectly (#10489)
```

File: changelog.txt [hunk 14]
```diff
@@ -1,3 +1,25 @@
...
+Fix #10486: [Script] Debug window requires AIs to be started before GS (#10487)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10486: Crash in debug window when GS started before AIs (#10487)
```

File: changelog.txt (hunk 2 above)
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10469: [Script] Negative numbers in League Table window were sorted incorrectly (#10471)
```

File: changelog.txt [hunk 15]
```diff
@@ -1,3 +1,25 @@
...
+Fix #10469, 5e14a20: [Script] League Table rating element is a int64 everywhere else (#10471)
+Fix #10465: Delay closing of network join progress window (#10466)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10465: Crash on timeout if user never enters a password for server (#10466)
```

File: changelog.txt [hunk 16]
```diff
@@ -1,3 +1,25 @@
...
+Fix #10461, Fix #10280: Town actions GUI issues (#10462)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10280, #10461: Crash on opening town windows as a spectator (#10462)
```

File: changelog.txt [hunk 17]
```diff
@@ -1,3 +1,25 @@
...
+Fix #10059: Clamp config item values to int, disallow negative random deviation, and allow values up to 12 digits including '-' (#10444)
```
Comment (LordAro) [inline changelog.txt]:
```suggestion
Fix #10059: Script config values stored in the config file could cause crashes (#10462)
```

**Review by LordAro — COMMENTED** [review]
Sorry, I have suggestions on almost every line :)

**Comment by 2TallTyler** [comment]
No need to apologize for having so many good suggestions! 😃

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**